haystack/connector/crawler.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import re
+from pathlib import Path
+from urllib.parse import urlparse
+from typing import List, Any, Optional, Dict, Tuple, Union
+from haystack.schema import Document, BaseComponent
+logger = logging.getLogger(__name__)
+
+
+class Crawler(BaseComponent):
+    """
+    Crawl texts from a website so that we can use them later in Haystack as a corpus for search / question answering etc.
+
+    **Example:**
+    ```python
+    |    from haystack.connector import Crawler
+    |
+    |    crawler = Crawler()
+    |    # crawl Haystack docs, i.e. all pages that include haystack.deepset.ai/docs/
+    |    docs = crawler.crawl(urls=["https://haystack.deepset.ai/docs/latest/get_startedmd"],
+    |                         output_dir="crawled_files",
+    |                         filter_urls= ["haystack\.deepset\.ai\/docs\/"])
+    ```
+    """
+
+    outgoing_edges = 1
+
+    def __init__(self, urls: List[str], output_dir: str, crawler_depth: int = 1,
```

Comment:
Do we want the `urls` parameter in `__init__`?

Reply:
I thought it might make sense when we think about the YAML of an indexing pipeline that (periodically) crawls a certain list of URLs (e.g. our haystack docs). Where would you define this in the YAML if it's not an instance attribute?

Reply:
Yes, exactly, that's the case I thought of as well. We can keep the parameter but possibly make it `Optional`.

Reply:
Ok makes sense